eman2/protocols/protocol_tomo_extraction.py

```diff
@@ -146,42 +144,50 @@ def readSetOfTomograms(self, workDir, tomogramsSet):
         for index in range(1, n + 1):
             subtomogram.cleanObjId()
             subtomogram.setLocation(index, workDir)
+            if self.downFactor.get() != 1:
+                I = xmippLib.Image(subtomogram.getLocation())
+                x, y, z, _ = I.getDimensions()
+                I.scale(int(x / self.downFactor.get()), int(y / self.downFactor.get()), int(z / self.downFactor.get()))
+                fnSubtomo = self._getExtraPath("downsampled_subtomo%d.mrc" % index)
+                I.write(fnSubtomo)
+                subtomogram.setLocation(fnSubtomo)
             subtomogram.setCoordinate3D(self.coordDict[index-1])
             subtomogram.setAcquisition(self.getInputTomogram().getAcquisition())
             tomogramsSet.append(subtomogram)
 
-    def createOutputStep(self):
 
+    def createOutputStep(self):
         suffix = self._getOutputSuffix(SetOfSubTomograms)
         self.outputSubTomogramsSet = self._createSetOfSubTomograms(suffix)
-        self.outputSubTomogramsSet.setSamplingRate(self.getInputTomogram().getSamplingRate())
+        self.outputSubTomogramsSet.setSamplingRate(self.getInputTomogram().getSamplingRate()*self.cshrink)
         self.outputSubTomogramsSet.setCoordinates3D(self.inputCoordinates)
-
         self.readSetOfTomograms(self._getExtraPath(pwutils.join('sptboxer_01', 'basename.hdf')),
                                 self.outputSubTomogramsSet)
-
         self._defineOutputs(outputSetOfSubtomogram=self.outputSubTomogramsSet)
         self._defineSourceRelation(self.inputCoordinates, self.outputSubTomogramsSet)
 
+
     def writeSetOfCoordinates3D(self):
         self.coordsFileName = self._getExtraPath(
             pwutils.replaceBaseExt(self.getInputTomogram().getFileName(), 'coords'))
         self.coordDict = []
-
         out = file(self.coordsFileName, "w")
         for coord3DSet in self.inputCoordinates.get().iterCoordinates():
             out.write("%d\t%d\t%d\n" % (coord3DSet.getX(), coord3DSet.getY(), coord3DSet.getZ()))
             self.coordDict.append(coord3DSet.clone())
-
         out.close()
 
     # --------------------------- STEPS functions -----------------------------
     def extractParticles(self):
-
-        args = '%s --coords %s --boxsize %d' % (
-            self.getInputTomogram().getFileName(), pwutils.replaceBaseExt(self.getInputTomogram().getFileName(), 'coords'),
-            self.inputCoordinates.get().getBoxSize())
-
+        TsTomo = self.getInputTomogram().getSamplingRate()
```

Comment:
This is a bit picky but not sure if TsTomo and TsCoord are particularly descriptive and in general variables should start with lower case

Reply:
Maybe also a comment regarding this logic with TsTomo and TsCoord can be helpful...

Reply:
solved in https://github.com/scipion-em/scipion-em-eman2/pull/33/commits/9c158b38a1db38591511411de61b3ca96e084186